### PR TITLE
DOCS-5916 adding contracts

### PR DESCRIPTION
**Pull request by wwarner-salesforce**
DOCS-5916 adding contracts

### CODE REVIEW

File: modules/ROOT/pages/about-my-applications.adoc [hunk 1]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
 A contract holds information about the contract with a client ID and client secret for the API.
 Each application lists a description, its URL, redirect URIs, grant types, and a usage dashboard.
 
-The usage information lets you set the verification duration. If not specified, the range is the last 8 days. The usage lists the total number of requests to the API, average latency, the error rate, and graphs for total requests, average latency, requests by HTTP status code, and the error percentage.
+The usage information lets you set the verification duration. If not specified,
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
The usage information enables you to set the verification duration. If not specified,
```

File: modules/ROOT/pages/about-my-applications.adoc [hunk 2]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
+the range is the last 8 days. The usage lists the total number of requests to
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
the range is the previous eight days. The usage lists the total number of requests to
```

File: modules/ROOT/pages/about-my-applications.adoc [hunk 3]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
+the API, average latency, the error rate, and graphs for total requests,
+average latency, requests by HTTP status code, and the error percentage.
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
requests by HTTP status code, and the error percentage.
```

Comment (valkyrie69) [reply modules/ROOT/pages/about-my-applications.adoc]: (average latency was repeated)

File: modules/ROOT/pages/about-my-applications.adoc [hunk 4]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
 
 The applications result when users click Request Access to establish
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]: applications result? The application is created? I don't know what "applications result" means. 
If you don't have an application to link to your API, you can create the application by clicking the *Request Access* menu. You can then click *Request Access* to establish a link between your application and API.

Comment (valkyrie69) [reply modules/ROOT/pages/about-my-applications.adoc]: That's my best guess. I'm not sure what's going on there.

File: modules/ROOT/pages/about-my-applications.adoc [hunk 5]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
 a link between an application and an API. Either the application already exists,
 or the user creates the application from the Request Access menu.
 
 Applications in My Applications are registered using API Manager in Anypoint Platform.
 
+Requesting access registers a client and creates an API contract between an API
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
When you request access, a client is registered and creates an API contract between an API
```

File: modules/ROOT/pages/about-my-applications.adoc [hunk 6]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
+and a client application.
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
and the client application.
```

File: modules/ROOT/pages/about-my-applications.adoc [hunk 7]
```diff
@@ -12,14 +12,28 @@ image::ex2-my-applications.png[]
...
+
+APIs that have been protected by a client ID enforcement policy require
```
Comment (valkyrie69) [inline modules/ROOT/pages/about-my-applications.adoc]:
```suggestion
APIs that are protected by a client ID enforcement policy require
```

File: modules/ROOT/pages/to-request-access.adoc [hunk 8]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
-APIs that have been protected by a Client ID enforcement policy require API consumers to create an application to hold the Client ID and Secret. This lets the app access the API as well as provide consumer analytics for an API developer.
+Requesting access registers a client and creates an API contract between an API
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]: When you request access, a client is registered and creates an API contract between an API

File: modules/ROOT/pages/to-request-access.adoc [hunk 9]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
+and a client application.
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
and the client application.
```

File: modules/ROOT/pages/to-request-access.adoc [hunk 10]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
+
+APIs that have been protected by a client ID enforcement policy require
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
APIs that are protected by a client ID enforcement policy require
```

Comment (valkyrie69) [reply modules/ROOT/pages/to-request-access.adoc]: I assume the repetition of this text is intentional since it's two different files. Good opportunity for an include.  :-)

File: modules/ROOT/pages/to-request-access.adoc [hunk 11]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
+application to access the API and provides consumer analytics for an API
+developer.
+
+API contracts are visible to API consumers in Exchange, and visible to API
+providers in API Manager.
+
+API providers can view, manage, and revoke these contracts with the steps in
+xref:api-manager::manage-client-apps-latest-task.adoc[Managing API Instance Contracts on API Manager].
 
 If terms and conditions are set for an asset, you are prompted to accept the terms before access can be granted.
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
If terms and conditions are set for an asset, you are prompted to accept the terms before access is granted.
```

File: modules/ROOT/pages/to-request-access.adoc [hunk 12]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
 
 . Click an API asset to view more details.
 . Click Request Access.
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
. Click **Request Access**.
```

File: modules/ROOT/pages/to-request-access.adoc [hunk 13]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
-. If you do not have an application you want to provide access for, click Create A New Application in the Request Access menu. See below for more information.
+. If you do not have an application you want to provide access for, click Create A New Application in the Request Access menu. See the following section for more information.
```
Comment (valkyrie69) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
. If you do not have an application to which to provide access, click **Request Access > Create A New Application**. See the following section for more information.
```

**Review by valkyrie69 — APPROVED** [review]
Approved with suggestions

File: modules/ROOT/pages/to-request-access.adoc (hunk 10 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Good point. It is intentional, and it is a good opportunity for an include...

File: modules/ROOT/pages/to-request-access.adoc (hunk 8 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Fixed.

File: modules/ROOT/pages/about-my-applications.adoc (hunk 4 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: I rewrote for clarity, and included this text. There was a lot of sloppy text in this one, including things I missed such as "average latency" being repeated. Thank you for a very helpful review!

**Review by kuntald — APPROVED** [review]
LGTM

File: modules/ROOT/pages/to-request-access.adoc [hunk 14]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
-APIs that have been protected by a Client ID enforcement policy require API consumers to create an application to hold the Client ID and Secret. This lets the app access the API as well as provide consumer analytics for an API developer.
+When you request access, the client application is registered and creates an
```
Comment (priyaayyar) [inline modules/ROOT/pages/to-request-access.adoc]: To implement an API, the client application must first request access. When the request is approved by the API, a contract is created between the client application and the API, and the client application is registered.

File: modules/ROOT/pages/to-request-access.adoc [hunk 15]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
...
+API contract between the API and the client.
 
-If terms and conditions are set for an asset, you are prompted to accept the terms before access can be granted.
+APIs that are protected by a client ID enforcement policy require
+applications to have a client ID and client secret. This enables the
```
Comment (priyaayyar) [inline modules/ROOT/pages/to-request-access.adoc]: APIs that are protected by a Client ID Enforcement Policy require client application to provide the client ID and client secret credentials.

*the client ID and client secret credentials are automatically created when the client application is registered.*

File: modules/ROOT/pages/to-request-access.adoc [hunk 16]
```diff
@@ -4,13 +4,25 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
-APIs that have been protected by a Client ID enforcement policy require API consumers to create an application to hold the Client ID and Secret. This lets the app access the API as well as provide consumer analytics for an API developer.
+When you request access, the client application is registered and creates an
+API contract between the API and the client.
```
Comment (priyaayyar) [inline modules/ROOT/pages/to-request-access.adoc]: creates a contract between the API and client application--not an API contract.

**Review by priyaayyar — APPROVED** [review]

File: modules/ROOT/pages/to-request-access.adoc (hunk 14 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Updated to use this text. Thank you!

File: modules/ROOT/pages/to-request-access.adoc (hunk 15 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Added this info.

File: modules/ROOT/pages/to-request-access.adoc (hunk 16 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Fixed.

File: modules/ROOT/pages/about-my-applications.adoc [hunk 17]
```diff
@@ -12,14 +12,30 @@ image::ex2-my-applications.png[]
...
-a link between an application and an API. Either the application already exists,
-or the user creates the application from the Request Access menu.
+The usage information enables you to set the verification duration. If not specified,
+the range is the previous eight days. The usage lists the total number of requests to
+the API, average latency, the error rate, and graphs for total requests,
+requests by HTTP status code, and the error percentage.
 
 Applications in My Applications are registered using API Manager in Anypoint Platform.
 
+To implement an API, the client application must first request access. When the
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/about-my-applications.adoc]: Incorrect statement. API can be created regardless of having one, many or no contracts. I would say instead of "To implement an API" something like "To register a client application to an existing API".

File: modules/ROOT/pages/about-my-applications.adoc [hunk 18]
```diff
@@ -12,14 +12,30 @@ image::ex2-my-applications.png[]
...
+request is approved by the API, a contract is created between the client
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/about-my-applications.adoc]: "request is approved by the API owner"

File: modules/ROOT/pages/about-my-applications.adoc [hunk 19]
```diff
@@ -12,14 +12,30 @@ image::ex2-my-applications.png[]
...
+application and the API, and the client application is registered.
+
+If you don't have an application to link to your API, you can create the application
+in the Request Access menu. You can then click Request Access to connect your
+application and the API.
+
+APIs that are protected by a client ID enforcement policy require client
+applications to provide a client ID and client secret. The client ID and client
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/about-my-applications.adoc]: client secret is optional

File: modules/ROOT/pages/about-my-applications.adoc [hunk 20]
```diff
@@ -12,14 +12,30 @@ image::ex2-my-applications.png[]
...
 
 Applications in My Applications are registered using API Manager in Anypoint Platform.
 
+To implement an API, the client application must first request access. When the
+request is approved by the API, a contract is created between the client
+application and the API, and the client application is registered.
+
+If you don't have an application to link to your API, you can create the application
+in the Request Access menu. You can then click Request Access to connect your
+application and the API.
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/about-my-applications.adoc]: This is a basic case, that can be all right. For your radar: If at least one SLA is defined in the API, you have to choose one when requesting access. Also, terms and conditions have to be accepted, if defined.

File: modules/ROOT/pages/about-my-applications.adoc [hunk 21]
```diff
@@ -12,14 +12,30 @@ image::ex2-my-applications.png[]
...
+
+APIs that are protected by a client ID enforcement policy require client
+applications to provide a client ID and client secret. The client ID and client
+secret credentials are automatically created when the client application is
+registered. These credentials enable the application to access the API and
+provide consumer analytics for an API developer.
+
+API contracts are visible to API consumers in Exchange, and visible to API
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/about-my-applications.adoc]: Not sure if this is true in general. If your are org owner, then yes. I believe (please double check) that only org owners and API owners have visibility over the contracts of a certain API.

File: modules/ROOT/pages/to-request-access.adoc [hunk 22]
```diff
@@ -4,13 +4,27 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
-APIs that have been protected by a Client ID enforcement policy require API consumers to create an application to hold the Client ID and Secret. This lets the app access the API as well as provide consumer analytics for an API developer.
+To implement an API, the client application must first request access. When the
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/to-request-access.adoc]: See previous comment.

File: modules/ROOT/pages/to-request-access.adoc [hunk 23]
```diff
@@ -4,13 +4,27 @@ include::_attributes.adoc[]
...
+request is approved by the API, a contract is created between the client
+application and the API, and the client application is registered.
 
-If terms and conditions are set for an asset, you are prompted to accept the terms before access can be granted.
+APIs that are protected by a client ID enforcement policy require client
+applications to provide a client ID and client secret. The client ID and client
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/to-request-access.adoc]: Secret is optional

File: modules/ROOT/pages/to-request-access.adoc [hunk 24]
```diff
@@ -4,13 +4,27 @@ include::_attributes.adoc[]
...
+secret credentials are automatically created when the client application is
+registered. These credentials enable the application to access the API and
```
Comment (nahuel-dallavecchia) [inline modules/ROOT/pages/to-request-access.adoc]: This is somewhat confusing since it seems to imply that only with this policy and this request analytics is enabled. Analytics is enabled for all tracked applications. In that case, they are generated even if no policy is applied.

**Review by nahuel-dallavecchia — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/about-my-applications.adoc (hunk 17 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: Fixed. Thank you!

File: modules/ROOT/pages/about-my-applications.adoc (hunk 18 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: Fixed.

File: modules/ROOT/pages/about-my-applications.adoc (hunk 19 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: Fixed.

File: modules/ROOT/pages/about-my-applications.adoc (hunk 20 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: This is good info. Added.

File: modules/ROOT/pages/to-request-access.adoc (hunk 22 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Fixed.

File: modules/ROOT/pages/to-request-access.adoc (hunk 23 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Fixed.

File: modules/ROOT/pages/to-request-access.adoc (hunk 24 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-request-access.adoc]: Fixed.

File: modules/ROOT/pages/about-my-applications.adoc (hunk 21 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: @nahuel-dallavecchia who can confirm this? Exchange developers?

Comment (nahuel-dallavecchia) [reply modules/ROOT/pages/about-my-applications.adoc]: Try with Samuel Centurion, QA Lead of API Manager first.

Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: Thank you, Nahuel! Looks like Sebastian Lamelas on the Exchange team already answered this one: "The API owner has visibility over API contracts in API Manager (visibility of ALL contracts ever created for that API instance), whereas the API consumer has visibility only of the API contracts created between an API instance and *his* application in "My Applications" page in Exchange."

Comment (wwarner-salesforce) [reply modules/ROOT/pages/about-my-applications.adoc]: Fixed.

**Review by nahuel-dallavecchia — APPROVED** [review]

File: modules/ROOT/pages/about-my-applications.adoc [hunk 25]
```diff
@@ -9,17 +9,37 @@ applications that hold a contract between an API and a user.
 
 image::ex2-my-applications.png[]
 
-A contract holds information about the contract with a client ID and client secret for the API.
-Each application lists a description, its URL, redirect URIs, grant types, and a usage dashboard.
+A contract holds information including a client ID and optional client secret
+for the API. Each application lists a description, its URL, redirect URIs,
+grant types, and a usage dashboard.
```
Comment (kbostrom) [inline modules/ROOT/pages/about-my-applications.adoc]: Does the contract hold information FOR the API or ABOUT the API?
Would it be just as accurate to say:
"A contract includes information about an API, including its client ID; its client secret, if any; its description; its URIs; its grant types, and its usage dashboard"?
Or are these two sentences about two different things: the contract that includes ID and secret, and the API that includes the description etc.?
Clarify.

File: modules/ROOT/pages/about-my-applications.adoc [hunk 26]
```diff
@@ -9,17 +9,37 @@ applications that hold a contract between an API and a user.
...
 
-The usage information lets you set the verification duration. If not specified, the range is the last 8 days. The usage lists the total number of requests to the API, average latency, the error rate, and graphs for total requests, average latency, requests by HTTP status code, and the error percentage.
-
-The applications result when users click Request Access to establish
-a link between an application and an API. Either the application already exists,
-or the user creates the application from the Request Access menu.
+The usage information enables you to set the verification duration. If not specified,
+the range is the previous eight days. The usage lists the total number of requests to
+the API, average latency, the error rate, and graphs for total requests,
+requests by HTTP status code, and the error percentage.
```
Comment (kbostrom) [inline modules/ROOT/pages/about-my-applications.adoc]: the range is the previous eight days. The usage lists total number of requests to
the API, average latency, error rate, and graphs for total requests,
requests by HTTP status code, and error percentage.

Comment (kbostrom) [reply modules/ROOT/pages/about-my-applications.adoc]: Why is each line its own paragraph (+)? 
I see this method, and then I also see a method in which the + appears only at the beginning of a new paragraph.
Would you mind sending me a separate Slack message or email to explain?

File: modules/ROOT/pages/to-request-access.adoc [hunk 27]
```diff
@@ -4,13 +4,29 @@ include::_attributes.adoc[]
...
+applications to provide a client ID and optional client secret. The client ID and client
+secret credentials are automatically created when the client application is
+registered.
+
+An API owner can use API Manager to view all API contracts ever created for an
+API instance. An API consumer can use Exchange's *My Applications* page to view
+the API contracts created between an API instance and that consumer's client
+applications.
+
+API providers can view, manage, and revoke these contracts with the steps in
```
Comment (kbostrom) [inline modules/ROOT/pages/to-request-access.adoc]:
```suggestion
API providers can view, manage, and revoke these contracts by following the steps in
```

File: modules/ROOT/pages/to-request-access.adoc [hunk 28]
```diff
@@ -4,13 +4,29 @@ include::_attributes.adoc[]
...
+xref:api-manager::manage-client-apps-latest-task.adoc[Managing API Instance Contracts on API Manager].
+
+If terms and conditions are set for an asset, you are prompted to accept the terms before access is granted.
```
Comment (kbostrom) [inline modules/ROOT/pages/to-request-access.adoc]: How are these related to a contract? Are we saying that if the contract includes terms and conditions...?

File: modules/ROOT/pages/to-request-access.adoc [hunk 29]
```diff
@@ -4,13 +4,29 @@ include::_attributes.adoc[]
...
 
 . Click an API asset to view more details.
-. Click Request Access.
-. If you do not have an application you want to provide access for, click Create A New Application in the Request Access menu. See below for more information.
+. Click **Request Access**.
```
Comment (kbostrom) [inline modules/ROOT/pages/to-request-access.adoc]: Here and below, why the double asterisks?   **   **

File: modules/ROOT/pages/to-request-access.adoc [hunk 30]
```diff
@@ -4,13 +4,29 @@ include::_attributes.adoc[]
...
+. If you do not have an application to which to provide access, click **Request Access > Create A New Application**. See the following section for more information.
```
Comment (kbostrom) [inline modules/ROOT/pages/to-request-access.adoc]: Suggest naming the section rather than saying "the following section," to keep the text independent of reshuffling.

**Review by kbostrom — APPROVED** [review]
Here you go!
Please feel free to query suggestions or to comment if they change your intended meaning, so that I might learn from you.
I have also asked a couple of questions that I would like Slack answers about (the use of + before every line vs. every para, and the use of double asterisks).
Thanks!

File: modules/ROOT/pages/about-my-applications.adoc (hunk 25 above)
Comment (nahuel-dallavecchia) [reply modules/ROOT/pages/about-my-applications.adoc]: A contract is the link between a client application and an API. An API can have only one contract with a specific client application. A client application can have several contracts with different APIs. The only thing a contract may hold is SLA tier info and terms and conditions. A client application is identified by a client ID and client secret (both mandatory and generated by the platform). A client application holds as well, description , grant types supported, etc. Usage (aka analytics) is stored in a different place.

File: modules/ROOT/pages/to-request-access.adoc (hunk 28 above)
Comment (nahuel-dallavecchia) [reply modules/ROOT/pages/to-request-access.adoc]: Terms & Conditions may be defined at organization level. If defined, then the contract stores the terms and conditions accepted when the contract was created.

File: modules/ROOT/pages/to-request-access.adoc (hunk 29 above)
Comment (nahuel-dallavecchia) [reply modules/ROOT/pages/to-request-access.adoc]: Please see: https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#formatted-text

File: modules/ROOT/pages/about-my-applications.adoc (hunk 26 above)
Comment (nahuel-dallavecchia) [reply modules/ROOT/pages/about-my-applications.adoc]: Please see: https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#breaks

**Comment by wwarner-salesforce** [comment]
@kbostrom and @nahuel-dallavecchia , thank you both very much for all your help here! I sent a Slack message to Kim explaining the + marks and double asterisks.
